### PR TITLE
Adding 6 hour cooldown to follow messages in chat

### PR DESCRIPTION
**Pull request by mja00**
Checks the last follow message sent, if it's more than 6 hours ago it'll send a new message, if the message doesn't exist it'll send a new one, otherwise it just silently follows the streamer.

### CODE REVIEW

File: lib/glimesh/chat.ex [hunk 1]
```diff
@@ -192,6 +192,20 @@ defmodule Glimesh.Chat do
...
+  Gets the latest follower message from a channel for a user
+  """
+  def get_follow_chat_message_for_user(channel, user) do
+    Repo.all(
+      from m in ChatMessage,
+      where: m.channel_id == ^channel.id and m.user_id == ^user.id and m.is_followed_message == true,
+      order_by: [desc: :inserted_at],
+      limit: 1
+    )
+    |> Enum.reverse()
```
Comment (clone1018) [inline lib/glimesh/chat.ex:206]:
```suggestion

```
Likely there's no need to reverse a list of one.

File: lib/glimesh/account_follows.ex [hunk 2]
```diff
@@ -97,4 +99,21 @@ defmodule Glimesh.AccountFollows do
   def list_following(user) do
     Repo.all(from f in Follower, where: f.user_id == ^user.id)
   end
+
+  defp no_follow_message_recently(channel, user) do
```
Comment (clone1018) [inline lib/glimesh/account_follows.ex:103]:
```suggestion
  defp sent_follow_message_recently?(channel, user) do
```

File: lib/glimesh/chat.ex [hunk 3]
```diff
@@ -192,6 +192,20 @@ defmodule Glimesh.Chat do
   """
   def get_chat_message!(id), do: Repo.get!(ChatMessage, id) |> Repo.preload([:user, :channel])
 
+
+  @doc """
+  Gets the latest follower message from a channel for a user
+  """
+  def get_follow_chat_message_for_user(channel, user) do
+    Repo.all(
```
Comment (clone1018) [inline lib/glimesh/chat.ex:200]: Could this be .one?

**Review by clone1018 — CHANGES_REQUESTED** [review]

File: lib/glimesh/account_follows.ex [hunk 4]
```diff
@@ -43,12 +44,13 @@ defmodule Glimesh.AccountFollows do
...
-    broadcast(results, :followers)
-
     channel = ChannelLookups.get_channel_for_user(streamer)
 
-    if !is_nil(channel) and Glimesh.Chat.can_create_chat_message?(channel, user) do
-      Glimesh.Chat.create_chat_message(user, channel, %{
+    broadcast(results, :followers)
+
+    if !is_nil(channel) and Glimesh.Chat.can_create_chat_message?(channel, user) and
+         no_follow_message_recently(channel, user) do
```
Comment (mja00) [inline lib/glimesh/account_follows.ex:52]:
```suggestion
         sent_follow_message_recently?(channel, user) do
```

File: lib/glimesh/chat.ex (hunk 3 above)
Comment (mja00) [reply lib/glimesh/chat.ex:200]: Yes :)
```suggestion
    Repo.one(
```
I wasn't fully sure how I was gonna do it, plus I did a fat copy-paste of the function higher up

**Pull request merged**